build-tools/packages/build-tools/src/type-test-generator/main.ts

```diff
@@ -62,25 +63,42 @@ const currentFile = new Project({
 
 const previousTsConfigPath = `${previousBasePath}/tsconfig.json`;
 let previousFile: SourceFile;
-if (existsSync(previousTsConfigPath)) {
-	const project = new Project({
-		skipFileDependencyResolution: true,
-		tsConfigFilePath: previousTsConfigPath,
-	});
+const project = new Project({
+	skipFileDependencyResolution: true,
+	tsConfigFilePath: existsSync(previousTsConfigPath) ? previousTsConfigPath : undefined,
+});
+// Check for existence of alpha and add appropriate file
+if (existsSync(alphaFilePath)) {
+	project.addSourceFilesAtPaths(alphaFilePath);
+	previousFile = project.getSourceFileOrThrow("<package-name>-alpha.d.ts");
+	// If alpha doesn't exist but index.ts and tsconfig.json do
+} else if (existsSync(previousTsConfigPath)) {
 	previousFile = project.getSourceFileOrThrow("index.ts");
```

Comment:
Interesting. I assumed this operated on the `d.ts` file, rather than the actual source. Pointing at the source seems weird to me. @tylerbutler do you have any idea what this case is for? It seems like the fallback case of looking at the `d.ts` files should do the same thing?

Reply:
Given that there are known cases where type checking against the source and d.ts produce different results, testing against the d.ts should probably be all we do as thats what users of the package get and will be more consistent when comparing local and published versions.